### ISSUE TITLE
Unhandled exception at 0x00413367 on Windows 10 x64

### ISSUE THREAD

**Issue by KoenLav** [perak/kitchen-site#241]
This error occurs irregularly when creating collections:

Unhandled exception at 0x00413367 in meteor-kitchen.exe: 0xC0000005: Access violation reading location 0x00000000.

Once meteor-kitchen has finished creating the collections, it never occurs, but at any point during the creation of collections this error may (or may not) occur. It doesn't occur at a specific collection.

**Comment by perak** [comment]
@KoenLav does that happens with official kitchen examples too? (e.g. example-dataview)? If not, please help me reproduce (.json which produces error).

**Comment by KoenLav** [comment]
I pinpointed the problem to 'use_collection2: true'.

If I adjust the dataviews sample to include 'collection2: true' the error occurs, if I set it to false the error disappears. But it only occurs on Windows (10 x64), not Ubuntu or Mac.

**Comment by perak** [comment]
@KoenLav I just tried to reproduce this issue on Windows 7 32-bit in VM - but everything works normally. 

Damn windows!

I'll need to install win10 x64 in VM...

(I hope I am doing the same thing as you do: example-dataview modified to `"use_collection2": true`, clean build into non-existing directory) ?

**Comment by perak** [comment]
(it's strange that program behaves differently in different versions of windows!? non-sense)

**Comment by KoenLav** [comment]
It's possible I ran it with collection2 false first (so it would not be in
an empty directory).

Other than that it's exactly right.
Op 15 feb. 2016 8:30 PM schreef "Petar Korponaić" <notifications@github.com

**Comment by perak** [comment]
@KoenLav is this exact steps:

1. run with collection2: false

2. run with collection2: true

?

**Comment by perak** [comment]
(worked that way too)

**Comment by KoenLav** [comment]
Yeah, but I think it will also occur when run on an empty directory, let me
check that.
Op 15 feb. 2016 8:58 PM schreef "Petar Korponaić" <notifications@github.com

**Comment by perak** [comment]
I'm trying to find bug "blind" by reviewing code (instead installing win10) to possibly save my time...

**Comment by KoenLav** [comment]
I'd be fine with looking at the code concerned with collection2 myself as
well, maybe I could be of help there.

2016-02-15 21:00 GMT+01:00 Petar Korponaić <notifications@github.com>:

**Comment by KoenLav** [comment]
I thought I was going crazy, I just had to do it 10 times before I got the error (on a filled directory).

It appears that the more collections the more likely it is that the error occurs (but it does not occur always).

**Comment by KoenLav** [comment]
With the dataviews example it occurs about once in every 5 tries, but with our own json it occurs 8/10 times.

**Comment by perak** [comment]
I got it! `example-invoices` crashes...

**Comment by KoenLav** [comment]
Great, we made your application crash! :tada: 

Thanks for your help :+1:

**Comment by perak** [comment]
But... with debugger on - it don't want to crash! :rage: ...

**Comment by perak** [comment]
Still no luck. I cannot reproduce. It crashed once but unfortunately without debugger / call stack. I tried hundreds of times, with/without collection2, created 10 collections, empty dir, non-empty dir...

**Comment by KoenLav** [comment]
Strange, can I do anything to help?

I'd be happy to take a look at some code snippets.

**Comment by perak** [comment]
Sorry, code is (still) closed. I just want to reproduce issue once, only once with debugger on - but it don't want to cooperate :)

**Comment by KoenLav** [comment]
You could try adding even more collections (we have about 10 in our JSON).

**Comment by perak** [comment]
Tried that too... giving up after 2 hours of trying :(

**Comment by KoenLav** [comment]
Yeah, I guess that's best, these issues are a PITA to troubleshoot... (When they don't occur consistently.)
